corgi/api/filters.py

```diff
@@ -210,6 +222,30 @@ def filter_active_streams(
             return queryset
         return queryset.active_streams(include=value)
 
+    @staticmethod
+    def filter_ofuri_components(
+        queryset: QuerySet[Component], name: str, value: str
+    ) -> QuerySet["Component"]:
+        """'latest' and 'root components' filter automagically turn on
+        when the ofuri parameter is provided"""
+        if value in EMPTY_VALUES:
+            return queryset
+        model, model_type = get_model_ofuri_type(value)
+        if isinstance(model, Product):
+            components_for_model = queryset.filter(products__ofuri=value)
+        elif isinstance(model, ProductVersion):
+            components_for_model = queryset.filter(productversions__ofuri=value)
+        elif isinstance(model, ProductStream):
+            components_for_model = queryset.filter(productstreams__ofuri=value)
+        elif isinstance(model, ProductVariant):
+            components_for_model = queryset.filter(productvariants__ofuri=value)
+        else:
+            components_for_model = queryset
```

Comment:
this is now just setting queryset - was unsure of embedding http behaviour here but yes I can otherwise the options are 1) throw error (as before) 2) return original queryset 3) return empty queryset ...